### PR TITLE
feat: type note:* and label:* webhook payloads

### PR DESCRIPTION
**Pull request by scottlovegrove**
## Summary

Follow-up to #586. Narrows two more slices of `WebhookPayloadSchema` and splits the test file by resource.

- **note:\***: new `Comment*` payload variants (wire names stay `note:*`). `eventData` is a union of `WebhookItemComment` (carries the embedded `Task` and a direct URL) and `WebhookProjectComment` (just `projectId`). Consumers narrow with `'taskId' in eventData`.
- **label:\***: new `Label*` payload variants. `WebhookLabelSchema` reuses `LabelSchema` — omitting `order` and re-adding it via a rename from the wire's `itemOrder` — so any future additions to the shared label shape flow through automatically. Surfaces `isDeleted`.
- **test layout**: the single `webhook-parser.test.ts` is split into domain-specific files next to their schemas (`tasks.test.ts`, `comments.test.ts`, `labels.test.ts`). Cross-cutting snake_case fixtures (`sign`, `rawTask`, `envelope`, `CLIENT_SECRET`) live in `src/test-utils/webhook-fixtures.ts`; resource-specific fixtures stay inline in their test file.

## Test plan

- [x] `npx vitest run` — 550 pass (13 new webhook tests across the three files)
- [x] `npx oxfmt --check src`
- [x] `npx tsc -p tsconfig.typecheck.json` (no new errors)
- [ ] Manual: register a webhook on a dev app, fire a `note:added` / `label:updated`, and confirm `parseWebhookPayload` returns the expected narrowed shape

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/types/webhooks/labels.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * Webhook label payloads differ from the REST label shape in two ways:
+ * they use `item_order` (not `order`) and always include `is_deleted`. This
+ * schema reuses {@link LabelSchema}, swapping `order` for `itemOrder` on
+ * the input side and renaming it back after validation, and surfaces
+ * `isDeleted` for downstream filtering.
+ */
+export const WebhookLabelSchema = LabelSchema.omit({ order: true })
+    .extend({
+        itemOrder: z.number().int(),
```
Comment (doistbot) [inline src/types/webhooks/labels.ts]: [P2] Maintain `.nullable()` for `itemOrder`

`LabelSchema` defines `order` as `z.number().int().nullable()`. To safely map `item_order` from the webhook payload without crashing if the backend delivers a `null` value, `itemOrder` should also be explicitly `.nullable()` here.

```suggestion
        itemOrder: z.number().int().nullable(),
```

File: src/types/webhooks/comments.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+/** A comment attached to a project, delivered in a webhook payload. */
+export type WebhookProjectComment = z.infer<typeof WebhookProjectCommentSchema>
+
+/**
+ * The `eventData` of a `note:*` webhook payload — either an item comment
+ * (includes `taskId` and an embedded {@link Task}) or a project comment
+ * (includes `projectId`). Narrow with `'taskId' in eventData`.
+ */
+export const WebhookCommentSchema = z.union([WebhookItemCommentSchema, WebhookProjectCommentSchema])
```
Comment (doistbot) [inline src/types/webhooks/comments.ts:92]: [P2] This union no longer enforces the existing comment invariant that exactly one of `itemId` or `projectId` is present. Because Zod object schemas strip unknown keys by default, a malformed payload containing both IDs will be accepted as an item comment instead of being rejected, which can misclassify `eventData` for consumers narrowing on `'taskId' in eventData`. Add the same mutual-exclusion check here that the shared comment schema already applies.

**Review by doistbot — COMMENTED** [review]
This PR expands webhook typing by introducing narrowed payloads for comments and labels while neatly reorganizing the test suite into domain-specific files. Reusing existing schemas ensures forward compatibility, and the structural split improves overall test maintainability. To ensure robust parsing, there are a couple of edge cases to address regarding null safety for the label's `itemOrder` field and enforcing mutual exclusion between `itemId` and `projectId` in comment payloads to prevent misclassification.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-sdk-typescript&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-sdk-typescript%2Fpull%2F587&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253A7cbc12d0-3bf6-11f1-98b4-3905e5525400%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3A7cbc12d0-3bf6-11f1-98b4-3905e5525400&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

**Review by rmartins90 — COMMENTED** [review]
✅
